src/core/server/http/http_server.test.ts

```diff
@@ -1409,13 +1410,93 @@ describe('setup contract', () => {
   });
 
   describe('#registerStaticDir', () => {
+    const assetFolder = join(__dirname, 'integration_tests', 'fixtures', 'static');
+
     test('does not throw if called after stop', async () => {
       const { registerStaticDir } = await server.setup(config);
       await server.stop();
       expect(() => {
         registerStaticDir('/path1/{path*}', '/path/to/resource');
       }).not.toThrow();
     });
+
+    test('returns correct headers for static assets', async () => {
+      const { registerStaticDir, server: innerServer } = await server.setup(config);
+
+      registerStaticDir('/static/{path*}', assetFolder);
+
+      await server.start();
+      const response = await supertest(innerServer.listener)
```

Comment:
Wasn't sure how deep we want to go in term of IT, so I added coverage for the basic functionalities.

Did not add a test to assert that when the filesystem file changes, the handler detects it and the etag is updated because we don't really have any scenarios were our static assets are going to be modified atm (or do we?), and because writing/updating a file from an IT was tedious, but I can do it if we feel like it's necessary.

Reply:
a bug might creep up in the future if we dont assert that the etag changes when the content does. Also do we expose this API to custom plugins? then this might be used outside the scope of this repo as well. IMO a json file is the easiest to create/update/delete for this case.

Reply:
Fair enough. Will add a test